tasks/workflow/dispatcher.go

```diff
@@ -81,34 +83,42 @@ func getNbAndMaxTasksWaitTimeMs() (float32, float64, error) {
 	return nb, max, nil
 }
 
-func (d *Dispatcher) emitTasksMetrics() {
+func (d *Dispatcher) emitMetrics() {
 	d.wg.Add(1)
 	go func() {
 		defer d.wg.Done()
 		lastWarn := time.Now().Add(-6 * time.Minute)
 		for {
 			select {
 			case <-time.After(time.Second):
-				metrics.SetGauge([]string{"workers", "free"}, float32(len(d.WorkerPool)))
-				nb, maxWait, err := getNbAndMaxTasksWaitTimeMs()
-				if err != nil {
-					now := time.Now()
-					if now.Sub(lastWarn) > 5*time.Minute {
-						// Do not print each time
-						lastWarn = now
-						log.Printf("Warning: Failed to get Max Blocked duration for tasks: %+v", err)
-					}
-					continue
-				}
-				metrics.AddSample([]string{"tasks", "maxBlockTimeMs"}, float32(maxWait))
-				metrics.SetGauge([]string{"tasks", "nbWaiting"}, nb)
+				d.emitWorkersMetrics()
+				d.emitTasksMetrics(&lastWarn)
 			case <-d.shutdownCh:
 				return
 			}
 		}
 	}()
 }
 
+func (d *Dispatcher) emitWorkersMetrics() {
+	metrics.SetGauge([]string{"workers", "free"}, float32(len(d.WorkerPool)))
+}
+
+func (d *Dispatcher) emitTasksMetrics(lastWarn *time.Time) {
```

Comment:
Is this still relevant?
Shouldn't we measure task executions instead? If so it would be interesting to do it asap when we are doing breaking changes.

Reply:
Which is the metric you ask about ? 
**yorc.workers.free** ?
I this its steel relevant ...

Reply:
Actually, I'm confused about what we are talking about.
Do we mesure tasks or tasks executions?

Reply:
I kept the following measurements about task executions : 

- number of tasks waiting in INITIAL state
- the highst waiting time

This is made possible by looking for tasks **status** and **creation date** in the consul KV. 
There are no such information about taskExecutions.

Reply:
I think it makes sense to add `creation date` to task executions

Reply:
To summarize, currently there are no more measures on _tasks_, only on _taskExecutions_.
Concerning _wait_ time, we measure the time waited by each _taskExecution_ before execution start and the _number of waiting_ taskExecutions.